pkg/explorer/elements/transaction.go

```diff
@@ -117,25 +123,27 @@ func (e *elements) GetTransactionsForAddress(addr string, blindingKey []byte) ([
 		txids := info["txids"].([]interface{})
 
 		txs := make([]explorer.Transaction, 0, len(txids))
-		chTxs := make(chan explorer.Transaction)
-		chErr := make(chan error, 1)
-
-		// TODO: this can be skipped as soon as the wallet pkg supports using
-		// blinders for blinding instead of forcing to unblind inputs with
-		// blinding private keys.
-		for _, txid := range txids {
-			go e.getTxDetails(txid.(string), chTxs, chErr)
-			select {
-			case tx := <-chTxs:
-				if tx != nil {
-					txs = append(txs, tx)
-				}
-			case err := <-chErr:
-				close(chTxs)
-				close(chErr)
-				return nil, fmt.Errorf("tx details: %w", err)
+		chRes := make(chan txResult)
+		wg := &sync.WaitGroup{}
+		wg.Add(len(txids))
+
+		go func() {
```

Comment:
My suggestion is, for the sake of readability is to have one anon func where we fill values in channel and close it with combination of wait group

````
go func() {
    defer close(chRes)
    for i := range txids {
	txid := txids[i]
	go e.getTxDetails(txid.(string), chRes, wg)
    }
    wg.Wait()
}
````

Reply:
I might agree on the purpose, but making such suggested change causes the InitWallet w/ restore to fail and also the e2e test to fail as well sometimes.
Not really sure why, but appraently the way you're suggesting seems  to be less efficient, slowing down the execution.

Therefore, I won't commit the change.
If it's just for the sake of readability, since we handle this kind of operations always in the same way, we could reason about encapsulating the code in a sort of helper. I would open a separated ticket for it though.